src/OperationProcessor.ts

```diff
@@ -67,64 +67,85 @@ export default class OperationProcessor {
    * @param operation The operation to apply against the given current DID Document (if any).
    * @param previousOperation The previously operation applied if any. Used for operation validation.
    * @param currentDidDocument The DID document to apply the given operation against.
-   * @returns undefined if any validation fails; the updated document otherwise.
+   * @returns [isOperationValid, updatedDidDocument]; isOperationValid is a boolean that indicates if the
+   *          operation is valid given the operation context. If the operation is valid, updatedDidDocument
+   *          contains the updated document, Otherwise, it contains currentDidDocument (unchanged).
```

Comment:
Nit: consider calling out that `updatedDidDocument` would be `undefined` in case of valid delete operation.

Reply:
Done.